Sources/Core/AWSClientRuntime/Signing/AWSSigV4Signer.swift

```diff
@@ -9,7 +9,89 @@ import AwsCommonRuntimeKit
 import ClientRuntime
 import Foundation
 
-public class AWSSigV4Signer {
+public class AWSSigV4Signer: ClientRuntime.Signer {
+    public typealias IdObj = Credentials
+
+    public func sign(
+        requestBuilder: SdkHttpRequestBuilder,
+        identity: Credentials,
+        signingProperties: ClientRuntime.Attributes
+    ) async throws -> SdkHttpRequestBuilder {
+        guard let isBidirectionalStreamingEnabled = signingProperties.get(
+            key: AttributeKeys.bidirectionalStreaming
+        ) else {
+            throw ClientError.authError(
+                "Signing properties passed to the AWSSigV4Signer must contain T/F flag for bidirectional streaming."
+            )
+        }
+
+        let signingConfig = try constructSigningConfig(identity: identity, signingProperties: signingProperties)
+
+        let unsignedRequest = requestBuilder.build()
+        let crtUnsignedRequest: HTTPRequestBase
+        if isBidirectionalStreamingEnabled {
+            crtUnsignedRequest = try unsignedRequest.toHttp2Request()
+        } else {
+            crtUnsignedRequest = try unsignedRequest.toHttpRequest()
+        }
+
+        let crtSignedRequest = try await Signer.signRequest(
+            request: crtUnsignedRequest,
+            config: signingConfig.toCRTType()
+        )
+        let signedRequest = requestBuilder.update(from: crtSignedRequest, originalRequest: unsignedRequest)
+        return signedRequest
```

Comment:
return directly rather than store in a variable that is unused and use a comment string to indicate that you are getting the signedRequest for readability

Reply:
Changed as suggested.